### ISSUE TITLE
Make menu font size configurable

### ISSUE THREAD

**Issue by freewee** [RealRaven2000/QuickFolders#394]
Hi
A long time ago (!), the menu size of QuickFolder (when clicking on a tab) was the same as the menu size of Thunderbird.
Since a certain version, the size of the QuickFolder menu is the same as the size of the tabs, for me very very large...I got used to it, but is it normal? I was waiting for years a correction for this, but nothing! This on Ubuntu 22.04 and Mint 21.2 Cinnamon, on my desktop and laptop.
Thank you

**Comment by RealRaven2000** [comment]
Thanks for the issue (I didn't know there was an issue like this). Are you talking about the popup menus below the QuickFolders tab?

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/2fcdb846-ac7f-4101-b05b-dc78a44b221f)


it's width is dynamic and depends on the length of your folder names...

**Comment by RealRaven2000** [comment]
Edit - I inserted a screenshot in the previous comment - you may have to click on the github link to see it. Can you also provide a screenshot in case it's something else?

**Comment by freewee** [comment]
Thank you for the interest!
I send a comparative picture: on the left, menu from the QuickFolder button: it is the same size as a menu from the toolbar of Thunderbird; on the right, menu from a tab:it is the same size as the tab. As I said, it was not like that with ealier versions of QuickFolder, months ago,the menus from the tabs haved the same size as those from Thunderbird. The "problem" appeared after an update of QuickFolder, and I always thought it will be corrected, so I didn't note the version...
![Capture d’écran du 2023-08-20 17-31-07](https://github.com/RealRaven2000/QuickFolders/assets/19634222/f8cfe5b5-0432-4c67-8213-fb1867fac544)

**Comment by RealRaven2000** [comment]
That's a good image., so first of all it looks like the font size is inherited from that of the button, here my production profile in Tb 102.14.0

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/00906844-f1ee-405b-a777-d2b97aa439e8)

So you have the option to decrease the font size of the tabs here on "Bling". It looks like you have a larger size than default (set to 0 for default size) - maybe try 11 or 12:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/5dffefe1-44fa-4e29-815f-c484e3ac0328)

**Comment by freewee** [comment]
Yes of course, if I decrease the tab font size, the size of the menu decrease, but the tabs become tiny.
Before, I could have big tabs with small menu (same size as the Thunderbird menus);This regression (for me!) came with an update of QuickFolder, a long time ago, I don't remember when!
If now size font of the menus are linked to the size of the tabs, I will do with this, but I would prefer the old behaviour, same menu whatever the size of the tabs!
Thank you

**Comment by RealRaven2000** [comment]
> Yes of course, if I decrease the tab font size, the size of the menu decrease, but the tabs become tiny. Before, I could have big tabs with small menu (same size as the Thunderbird menus);This regression (for me!) came with an update of QuickFolder, a long time ago, I don't remember when! If now size font of the menus are linked to the size of the tabs, I will do with this, but I would prefer the old behaviour, same menu whatever the size of the tabs! Thank you

I think that there may be 2 schools of thought here, one for people with accessibility issues who find some fonts hard to see and the other one who would like to fit more items into their menu. One way I could achieve this in a gentle way would be maybe by adding a multiplier setting (1.0em / 0.9em / 0.8em etc) for the menu font, so you can still scale it down based on the parent element (the tab) ?

However, just thinking this through I also see an inconsistency here with the system menus (for the tool popups of elements on the left)

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/6b6456df-863d-4a2d-b7a6-d590653fecb9)

So even better might be an absolute setting (in pixels) for the font on all menu items... I am also seeing a problem in the "Apple Pills" theme which currently disables setting the font size for no reason.

**Comment by RealRaven2000** [comment]
> Yes of course, if I decrease the tab font size, the size of the menu decrease, but the tabs become tiny. Before, I could have big tabs with small menu (same size as the Thunderbird menus);This regression (for me!) came with an update of QuickFolder, a long time ago, I don't remember when! If now size font of the menus are linked to the size of the tabs, I will do with this, but I would prefer the old behaviour, same menu whatever the size of the tabs! Thank you

Sure thing - I can make a change but it's best if it is good for _all_ users not just you personally, so I think adding a new option for menu fonts and having it consistent across all menus will be best for all users. I will have the same behavior as with the tab font size, with a default of 0 which will set it to the same (absolute) default size of 11px:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/37f3e783-6de2-409c-bf18-a108ea4521ac)

**Comment by RealRaven2000** [comment]
While working on the implementation I interestingly found this code snippet:

```js
// =============
// MENU FONT SIZE 
// [issue 329] inconsistent menu font size
let newFontSize = prefs.ButtonFontSize || 12; // default value, 0!
styleEngine.setElementStyle(ss, "#QuickFolders-FoldersBox .QuickFolders-folder-popup * > label", "font-size", newFontSize + "px");
styleEngine.setElementStyle(ss, "#QuickFolders-Category-Box popupset * > label", "font-size", newFontSize + "px");
```

so it looks like I urgently need to reference issue #329 which also messes with menu font sizes. But I think the change you observed may also be a lot older.

**Comment by RealRaven2000** [comment]
Here is a preview version (still need to translate the options interface). I think partly the old selector which contained the sequence `* > label` didn't work. IIRC I used this to increase the specificity.

Thunderbird 102.14 and older: [QuickFolders-mx-5.18pre6.zip](https://github.com/RealRaven2000/QuickFolders/files/12395026/QuickFolders-mx-5.18pre6.zip)

Thunderbird 115: [QuickFolders-mx-6.1.2pre23.zip](https://github.com/RealRaven2000/QuickFolders/files/12395169/QuickFolders-mx-6.1.2pre23.zip)

----

To test the versions above, download the zip file, drag into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by freewee** [comment]
Thank you!
The menu from tabs is now small (slightly too small,!!!) but there is no change when I choose an other value than "0" in the menu option!

**Comment by RealRaven2000** [comment]
> Thank you! The menu from tabs is now small (slightly too small,!!!) but there is no change when I choose an other value than "0" in the menu option!

ok, maybe my selectors still don't work. default size is 12px now. I shall see if / how I can fix it.

**Comment by RealRaven2000** [comment]
> > Thank you! The menu from tabs is now small (slightly too small,!!!) but there is no change when I choose an other value than "0" in the menu option!

which version of Thunderbird?

**Comment by RealRaven2000** [comment]
don;t know why, the selector shows up in inspector, but the rule is empty:
![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/e61bb589-e05d-483a-b33b-7d1dce640033)

**Comment by RealRaven2000** [comment]
I think I need to split these rules up into separate blocks, this is what I had:

```css
#QuickFolders-FoldersBox .QuickFolders-folder-popup menu label,
#QuickFolders-FoldersBox .QuickFolders-folder-popup menuitem label {
  font-size: 12px;
}
```

 I believe I need to split this up into 2 rules so that I can overwrite them individually with JavaScript...

**Comment by freewee** [comment]
which version of Thunderbird?
102.13.0 (64 bits)

**Comment by RealRaven2000** [comment]
ok found 2 problems, here is a new version. You may have to restart Thunderbird because style sheets are cached and don't fully refresh when you load an Add-on with a changed style sheet:

[QuickFolders-mx-5.18pre11.zip](https://github.com/RealRaven2000/QuickFolders/files/12396565/QuickFolders-mx-5.18pre11.zip)

With this I can configure button font and menu size independently, on all QuickFolders toolbars:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/dfa5acab-a02c-4b4d-970c-81038d6d076a)

**Comment by freewee** [comment]
OK this time!
Thank you for your job and reactivity!

**Comment by RealRaven2000** [comment]
I added some padding proportional to the font-size, to make the button look less "crammed". After fixing this, there was still a major problem with menupopups, which inherited the font-size from the main button:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/c4f49025-9fb2-4438-b4ac-02e473f2b216)

So I added a rule overwriting font-size:inherit for the injected chevron; result is a constant height for all menu items, see:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/f62d4cff-a026-4e79-82c1-1e187887456f)

With a slightly smaller menu font size of 11px, I am wondering whether we should allow for tweaking the vertical space between subfolders to make it even narrower to allow for more subfolders? Here is a comparison with that font size, with a system menu (Tb Tools) beside it for comparison:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/22ee2a9d-9ce7-4dc5-987b-b2d137908105)

on my 2.5k screens I can fit 35 subfolders vertical (underneath the 4 command entries _download all, open in new tab, QF Command, Mail Commands_)

Test version below:

[QuickFolders-mx-6.1.2pre35.zip](https://github.com/RealRaven2000/QuickFolders/files/12419114/QuickFolders-mx-6.1.2pre35.zip)

**Comment by freewee** [comment]
Hi
I saw what you're talking about (items that takes up too much vertical space), but I didn't dare tell you that, I thought it was just my personal aesthetic feeling!!!
I will try the last build!
Thank you!

**Comment by freewee** [comment]
mx-6.1.2pre35.zip not compatible with TB 102!

**Comment by RealRaven2000** [comment]
> mx-6.1.2pre35.zip not compatible with TB 102!

that's right. I might backport some of these changes, but it is wasting a lot of resources...

I suggest to move on, Thunderbird 115 is just around the corner, and will be pushed to about 10% of the users (2,000,000 people) within the next 2 weeks. Either upgrade your current profile or install a beta version, I have been testing with betas since April. The code changes are huge, but I am very happy with the current results.

**Comment by RealRaven2000** [comment]
> Hi I saw what you're talking about (items that takes up too much vertical space), but I didn't dare tell you that, I thought it was just my personal aesthetic feeling!!! I will try the last build! 

I have opened #396 for compact density support. 
Anyway all good the menu font changes have been backported (I think).
but OUCH, now this happens in ESR102:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/cf679f31-146d-4abc-a3d6-130cb91e5718)

**Comment by RealRaven2000** [comment]
There is probably a problem with flex model in 102, the tabs misbehave now:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/45130622-3e7d-4f09-a7c1-dd645f5c11ef)

**Comment by RealRaven2000** [comment]
Fixed it (I accidentally saved quickfolders-layout.css in the wrong folder while messing with inspector) - here is the ESR102 version:

[QuickFolders-mx-5.18pre14.zip](https://github.com/RealRaven2000/QuickFolders/files/12421415/QuickFolders-mx-5.18pre14.zip)

---

To test the version above, download the zip file, drag into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by freewee** [comment]
I don' t think the menu is compact, I put 2 menus on the same picture!
![Capture d’écran du 2023-08-23 18-42-18](https://github.com/RealRaven2000/QuickFolders/assets/19634222/640cb7a9-d3e1-4482-bcf7-f2a6f099c3fc)

**Comment by RealRaven2000** [comment]
> I don' t think the menu is compact, I put 2 menus on the same picture!

did you restart Thunderbird? Style sheets are cached.

**Comment by RealRaven2000** [comment]
Also compact density in Tb102 is here:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/78f34d81-1a8d-4f05-bd5e-1c708eceed3e)

**Comment by RealRaven2000** [comment]
Here is a screenshot (windows 10) with pre14 and Compact Density enabled and menu font size 11:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/60debccf-d204-4912-a1eb-cd693081e882)

**Comment by freewee** [comment]
Compact Density enabled and menu font size 11,Mint 21.2 cinnamon
![Capture d’écran du 2023-08-23 20-30-36](https://github.com/RealRaven2000/QuickFolders/assets/19634222/394a992f-d413-4c4d-9cc7-a96bc597680c)

**Comment by RealRaven2000** [comment]
> Compact Density enabled and menu font size 11,Mint 21.2 cinnamon

If that's Linux, I may have to check on Ubuntu. there may be some specific styles I need to override.

Test versions for myself (just so I can transmit to the VM) - 
- [QuickFolders-mx-5.18pre21.zip](https://github.com/RealRaven2000/QuickFolders/files/12429734/QuickFolders-mx-5.18pre21.zip)
- [QuickFolders-mx-5.18pre22.zip](https://github.com/RealRaven2000/QuickFolders/files/12429962/QuickFolders-mx-5.18pre22.zip)

**Comment by RealRaven2000** [comment]
Ok, I checked on Ubuntu and tweaked the settings for compact spacing. Here is screenshot in Windows with Tb115, at the default font size of 12px (both for menus and the Thunderbird UI) - I highlighted messages 1 and 11 in the following screenshots.

Latest Versions, corrected & based on menu font size:
- Tb102: [QuickFolders-mx-5.18pre22.zip](https://github.com/RealRaven2000/QuickFolders/files/12430095/QuickFolders-mx-5.18pre22.zip)
- Tb115: [QuickFolders-mx-6.2pre53.zip](https://github.com/RealRaven2000/QuickFolders/files/12430100/QuickFolders-mx-6.2pre53.zip)

The rule is one-size fits all:

```css
:root[uidensity="compact"] .QuickFolders-folder-popup menuitem,
:root[uidensity="compact"] .QuickFolders-folder-popup menu {
  min-height: 1.1rem;
  padding-block: 0.112rem;
}
```

Here is a screenshot in Ubuntu with Tb 102.14, with Thunderbird **default font-size** (which seems to calculate as 14.66), looks like this:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/7bc60312-fbed-44e4-8c1f-3cd13f2963a9)

If I crank up the menu font size to 15px (still Ubuntu) we should see similar spacing:
![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/de27616f-b746-4f0f-bb54-aaa1ddb69bff)

Here is a comparison, with same version in Windows (I have 125% scaling going on there as my eyes got worse over the years writing code):

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/bc65df4a-e5df-4a61-802b-2465bdba8809)

However, spacings work out a little wider in Thunderbird 115 for some reason; so I may have to tweak a little there:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/747183ab-5e96-45aa-a16d-9ef07be3ceb8)

...here we only get 9 entries against 10 if we're using the same font size, but scales down ok with smaller fonts.

**Comment by freewee** [comment]
Hi thank you
I tested under Ubuntu, it is the same as Linux Mint Cinnamon.
Second and third line from the tab menu is upper than the others.
Font menu 14 TB 112 ; I test the TB 115 but buggy for me with Mint...
![Capture d’écran du 2023-08-24 16-27-56](https://github.com/RealRaven2000/QuickFolders/assets/19634222/ede0af36-5944-427b-ad82-57801e81d186)

**Comment by RealRaven2000** [comment]
Ok that is weird I know what you mean but it doesn't happen to me in Ubuntu

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/f6eb8716-3d5a-4c8f-a602-66ace8776991)

(I hope you restarted Thunderbird, as it caches style sheets)
Can you also check whether other popup menus (of folders that have children) are higher than the menuitems that represent folders without child folders:
![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/108eaa4a-24c2-4cff-b101-b5439212cdb5)


You can check the computed height own any popup line via inspector (Ctrl+Shift+i) by switching on the option "Disable Popup Auto-Hide" like this:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/06d498c7-c000-4705-a0c4-80d8f83807e5)

this will then leave any menu on screen for inspection:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/0d34d8f8-6223-4f8b-ba93-3ae7361674e6)

in Computed styles you can check the padding-block-start and padding-top rules - the winning rule is always on top.
![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/b854cf40-5140-49de-a9eb-6573e96b3f42)

A screenshot of the "layout" tab may also be helpful:
![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/6ac614f8-b22d-454d-88fc-6447ed65043d)

you can see in my case the height of a line is 20px and the padding works out as 1.64px

**Comment by freewee** [comment]
I will compare different things, I 'm in multiboot W7, W10 ,Ubuntu, Mint cinnamon version 21.2 and old 20.3! With separate profiles for Thunderbird.

**Comment by freewee** [comment]
I test my differents OS (I forgot Mate in my list of multiboot!), but I have the same behavior for all...

**Comment by freewee** [comment]
The problem is effectively with the items that have child. I will look at what you are asking to me!

**Comment by freewee** [comment]
![Capture d’écran du 2023-08-24 22-11-12](https://github.com/RealRaven2000/QuickFolders/assets/19634222/efa174a7-5fe5-4bc3-9bb8-9ca7fa273d65)

**Comment by RealRaven2000** [comment]
Thanks that helps. Its not the padding but the actual element being 30px high. Can you find out the computed font size? You can drill down into the label element also it may have unnecessary padding or margin. Inspector is a powerful tool but such investigations take some time and understanding. It can either be the label element or the ::after pseudoelement which injects the > chevron. Something takes up more space than it should…

**Comment by RealRaven2000** [comment]
Other attributes that have an influence on height:

```css
min-height
line-height
height
font-size
padding-block
margin-block
```

since i am using rem as units for `min-height` this is dependent on the font-size of the element. If that’s not set for label or menu it will be inherited from the parent element (eventually arriving at the `toolbarbutton`)

**Comment by freewee** [comment]
Hi
Computed font size is 26 px

**Comment by RealRaven2000** [comment]
> Computed font size is 26 px

thanks - on which element though, the menu element? Any chance you could hop on a quick Zoom Call, that would be faster...

**Comment by RealRaven2000** [comment]
- if you have time for a 15min Zoom  call over the next 2.5 hours, I am up for it, how can I DM you details? You can reach me at axel.grude@gmail.com  if you like

**Comment by freewee** [comment]
About the element, it was this one
![Capture d’écran du 2023-08-25 12-55-06](https://github.com/RealRaven2000/QuickFolders/assets/19634222/b9942462-b5d8-4d80-b62a-a0df2ab47982)
Unfortunately, I have "urgent" things to manage for the moment, I look at my PC sometime briefly!

**Comment by RealRaven2000** [comment]
Implemented in 6.2 - Published 30/08/2023 
also in 5.18 published 27/08/2023
